compiler/src/test/java/com/squareup/anvil/compiler/TestUtils.kt

```diff
@@ -356,6 +357,15 @@ fun Class<*>.contributedProperties(): List<KClass<*>>? {
     .filterIsInstance<KClass<*>>()
 }
 
+/**
+ * Assumes that there is only one constructor. For MembersInjector factories in particular, we
+ * make their constructor private to match behavior in Dagger 2.57, but under the hood Kotlin
+ * generates a public synthetic constructor with additional parameters that we must filter out.
+ */
+internal fun Class<*>.primaryConstructor(): Constructor<*> {
```

Comment:
Not sure if this is the best name. When reading `primaryConstructor()` I think of Kotlin's primary constructor. But this isn't necessarily what this function returns. Maybe `singleConstructor()`? Not the end of the world, though.

Reply:
That's fair, `singleConstructor` sounds like a good alternative to me